doc/cabal-commands.rst

```diff
@@ -121,7 +121,7 @@ Arguments and flags common to some or all commands are:
         $ cabal install --allow-newer=foo:base,lens --allow-newer=bar:time
 
     Finally, one can enable :option:`--allow-newer` permanently by setting
-    ``allow-newer: True`` in the ``~/.cabal/config`` file. Enabling
+    ``allow-newer: True`` in the ``~/.config/cabal/config`` file. Enabling
```

Comment:
Yes, I made the subjective decision to use the default path because it looks cleaner.  This is line with the existing documentation, which also says `~/.cabal/config` instead of `$CABAL_DIR/config`.  I think the vast majority of people will use the default config file location, and the minority who customises their XDG variables will know where to look when the documentation says `~/.config`.

Reply:
I'm not worried for those who customized XDG vars. But for those who will have preexisting ~/.cabal (which will be used instead of ~/.config) — probably, a vast majority of users. At least in the near future after we release this feature. Does it make sense?

Reply:
That is a problem, but it wouldn't be solved by using XDG variables in the documentation.  The only completely unambiguous solution is to repeat (or link to) the backwards compatibility rules whenever we mention the config file, but that would perhaps get quite tedious very quickly.

No matter what, I think we will have to suffer through a period where the location of the cabal-install config file will be a bit fuzzy to most users.  Hopefully most will likely not need to touch the config file at all.  The only truly dreadful situation will occur when novices use multiple versions of cabal-install simultaneously...

Reply:
Right. My view is that it'd be more robust to say "Cabal config file" instead of "./config/cabal/config" here and in couple similar places. And the phrase "Cabal config file" could be a hyperlink to a section explaining the file and where you typically find it.

Reply:
That is possible in the manual, but not in the CLI help text.  And I expect most novices to read the CLI help text, but not the manual.

Reply:
That fair. But the current thread, for example, is under a page of the manual.

Reply:
Alright, I fixed this everywhere in the manual.